### PR TITLE
Add 32 bit encoding and decoding capabilities

### PR DESCRIPTION
**Pull request by ktmf01**
This is still a work in progress. Decoding and encoding capabilities will probably be merged separately.

### CODE REVIEW

File: include/FLAC/format.h [hunk 1]
```diff
@@ -282,14 +282,24 @@ extern FLAC_API const char * const FLAC__SubframeTypeString[];
...
+	FLAC__VERBATIM_SUBFRAME_DATA_TYPE_INT64 /**< verbatim subframe has 64-bit int */
+} FLAC__VerbatimSubframeDataType;
+
 
 /** VERBATIM subframe.  (c.f. <A HREF="../format.html#subframe_verbatim">format specification</A>)
  */
 typedef struct {
-	const FLAC__int32 *data; /**< A pointer to verbatim signal. */
+	union {
+		const FLAC__int32 *int32; /**< A FLAC__int32 pointer to verbatim signal. */
```
Comment (sezero) [inline include/FLAC/format.h:299]: Bad choice of member name: many C sources out there define `int32` and/or `int64` types for themselves.

Comment (sezero) [reply include/FLAC/format.h:299]: `data32` and `data64` maybe???

File: include/FLAC/ordinals.h [hunk 2]
```diff
@@ -82,4 +82,20 @@ typedef FLAC__uint8 FLAC__byte;
...
+ * 64 bit signed int or 32 bit signed int */
+
+typedef enum {
+        FLAC__FLEX_INT_POINTER_DATA_TYPE_INT32,
+        FLAC__FLEX_INT_POINTER_DATA_TYPE_INT64
+} FLAC__FlexIntPointerDataType;
+
+typedef struct {
+        union {
+                const FLAC__int32 *int32;
```
Comment (sezero) [inline include/FLAC/ordinals.h:95]: same as above

File: include/FLAC/format.h (hunk 1 above)
Comment (ktmf01) [reply include/FLAC/format.h:299]: Thanks for the feedback. Do you think the general idea (having a union in a struct with a member indicating which union member is being used) uses the right concept for the job here?

Comment (sezero) [reply include/FLAC/format.h:299]: A simple `void*` pointer cast to correct type where it is used would have done the job.

Comment (ktmf01) [reply include/FLAC/format.h:299]: Thanks for the suggestion. I'll try and see whether that makes the code cleaner or not.

**Comment by ktmf01** [comment]
I just rebased because I moved the changes for CI to the right commits. Also, I moved from FlexIntPointer to simply casting to void*, as suggested

**Comment by ktmf01** [comment]
@ValZapod would you perhaps be willing to test this PR? You requested this functionality in #245

**Comment by ValZapod** [comment]
Well, I would not be a be to say anything past the fact: oh, yes, it is lossless in all 32 bits, not 24 bits in 32 bits! And to compare wav in beyond compare or ffmpeg md5...

**Pull request merged**